src/main/scala/nl.knaw.dans.easy.deposit/docs/dm/Author.scala

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit.docs.dm
+
+import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.{ SchemedKeyValue, SchemedValue, _ }
+
+case class Author(titles: Option[String] = None,
+                  initials: Option[String] = None,// TODO add dots in toString if none provided?
+                  insertions: Option[String] = None,
+                  surname: Option[String] = None,
+                  role: Option[SchemedKeyValue] = None,
+                  ids: Option[Seq[SchemedValue]] = None, // TODO xml
+                  organization: Option[String] = None,
+                 ) extends Requirements {
+  private val hasMandatory: Boolean = organization.isProvided || (surname.isProvided && initials.isProvided)
+  private val hasRedundant: Boolean = !surname.isProvided && (titles.isProvided || insertions.isProvided)
+  private val incompleteMsg = "Author needs one of (organisation | surname and initials)"
+  private val redundantMsg = "Author has no surname so neither titles nor insertions"
+  require(hasMandatory, buildMsg(incompleteMsg))
+  require(!hasRedundant, buildMsg(redundantMsg))
+
+  def isRightsHolder: Boolean = role.exists(_.key == "RightsHolder")
+
+  override def toString: String = { // TODO ID's for rightsHolders
+    def name = Seq(titles, initials, insertions, surname)
+      .collect{case Some(s) if s.trim.nonEmpty => s}
+      .mkString(" ")
+
+    (surname.find(_.trim.nonEmpty), organization.find(_.trim.nonEmpty)) match {
```

Comment:
@rvanheest with your way I end up with two ways of treating a `Some("   \t\n")`
* `.isProvided`
* filter/collect `.trim.nonEmpty`

Reply:
And I'm also using the approach below to turn Some's with empty strings into None's. This approach can become elaborate and easy to miss when a field is added some day.
https://github.com/DANS-KNAW/easy-deposit-api/blob/ec1c8a08c2eb0d3183e811a18ca8fd780e394437/src/main/scala/nl.knaw.dans.easy.deposit/docs/dm/Relation.scala#L55-L58